persistence/OrderPersistence.js

```diff
@@ -28,12 +28,15 @@ class OrderPersistence extends Persistence {
         this.assertExists(order, orderId, Order.entityName);
         await order.destroy();
     }
-    
+
     async getByProductId(productId) {
         const orders = await Order.findAll({ where: { productId } });
-        this.assertExists(orders, productId, Order.entityName);
-        return orders;
+        if (orders.length) {
+            this.assertExists(orders, productId, Order.entityName);
```

Comment:
As per my last comment, you are passing an array of Orders and the Id of the product?

The error message will be "Order with ID 'productId' doesn't exist" which is a complete lie.
Also we don't want to check if the orders exist here in the first place. Why? Because we want them to be defined or undefined in the Service class so we can throw an exception there. Now if someone tries to delete a product WITHOUT associated orders they will get 404? This will mean that the product can never get deleted.

Also you can never reach the check in the Service class since you will throw an exception here always.